scripts/build_twilio_library.py

```diff
@@ -52,14 +53,27 @@ def run_openapi_generator(parent_dir: str, to_generate: str, output_path: str, f
     print(f'Code generation completed at {output_path}')
 
 
-def get_domain_info(oai_spec_location: str, domain: str, is_file: bool = False) -> Tuple[str, str, str]:
+def get_domain_info(oai_spec_location: str, domain: str, is_file: bool = False, language: str = "") -> Tuple[str, str, str]:
     full_path = oai_spec_location if is_file else os.path.join(oai_spec_location, domain)
     parts = re.split(r'twilio_(.+?)_?(v\d+)?\.', domain, flags=re.IGNORECASE)
     domain_name = parts[1]
     api_version = parts[2] or ''
+    # added logic to fetch the domain name from servers url in spec file, instead for relying on file name
+    if language == 'java' and full_path.endswith('.json'):
```

Comment:
Based on L57 above, this means the logic below only gets fun when passing in a specific OAI file. But shouldn't we always want to pull the domain_name from the spec file?

Reply:
We added conditionality for java because currently go is fetching the domain_name from file name, and hence has a different directory structure and we did not want to impact that(this would need additional investigation on the possible impact for go). The check for json is because librarian passes json files only, so for java all json files will be included.